components/src/components/layout/viewport/Viewport.tsx

```diff
@@ -38,7 +38,8 @@ export const Viewport = forwardRef<HTMLDivElement | null, ViewportProps>(
       <>
         <StyledViewport ref={ref} {...props}>
           <RenderLayout>{children}</RenderLayout>
-          <Badge />
+          {data.capabilities?.badgeVisibility &&
```

Comment:
`badgeVisibility` indicates whether or not the Schematic customer (aka account) can hide the badge or not, correct?

If so, shouldn't this logic be such that if `badgeVisibility` is false, then we _always_ show <Badge />, otherwise we show it based on whether `settings.badge?.visibility === "hidden"`?

I'm not sure the best way to code such logic inline, maybe a new small component makes sense, like

```
export const SchematicBadge = () => {
    const { data, settings } = useEmbed();

    if (!data.capabilities?.badgeVisibility) {
      return <Badge />
    }

   if (settings.badge?.visibility !== "hidden") {
      return <Badge />
   }

   return <></>
}
```